src/buffer/out/textBuffer.cpp

```diff
@@ -1194,15 +1200,19 @@ const COORD TextBuffer::GetWordEnd(const COORD target, const std::wstring_view w
     // NOTE: the end anchor (this one) is exclusive, whereas the start anchor (GetWordStart) is inclusive
 
     // Already at the end. Can't move forward.
-    if (target == GetSize().EndExclusive())
+    if (limit && GetSize().CompareInBounds(target, *limit, true) >= 0)
+    {
+        return target;
+    }
+    else if (target == GetSize().EndExclusive())
     {
         return target;
     }

```

Comment:
this isn't needed anymore

Reply:
Wait what isn't? Is it already gone?

Reply:
_now_ it is. There was a nice way to combine both if branches because `CompareInBounds(_, _, true)` has support for `EndExclusive`.